### PR TITLE
Memory write fix

### PR DESCRIPTION
**Pull request by mdurrant-b3**
This fixes the problem where we don't write all partitions.   I'm not a fan but it works.  Opening so I can get tests run.

### CODE REVIEW

**Comment by dougwig** [comment]
check

File: acos_client/v21/action.py [hunk 1]
```diff
@@ -35,3 +35,13 @@ def reload(self, write_memory=False, **kwargs):
         write_memory = 1 if write_memory else 0
         return self._post("system.action.reload",
                           params={"write_memory": write_memory}, **kwargs)
+
+    def write_active(self, shared_partition=None, tenant_partition=None, **kwargs):
```
Comment (dougwig) [inline acos_client/v21/action.py]: This should just take one arg, "partition". If you need higher-level logic, like shared+tenant, that should happen up above.

File: acos_client/v21/action.py [hunk 2]
```diff
@@ -35,3 +35,16 @@ def reload(self, write_memory=False, **kwargs):
         write_memory = 1 if write_memory else 0
         return self._post("system.action.reload",
                           params={"write_memory": write_memory}, **kwargs)
+
+    def write_active(self, partitions=[], **kwargs):
```
Comment (dougwig) [inline acos_client/v21/action.py]: I'm not feeling this array interface. As a standalone library, it's awkward for the common case. For an array, a list comprehension by the caller is similar in complexity.

Comment (mdurrant-b3) [inline acos_client/v21/action.py]: Cedric had similar feelings.  Rewriting, update soon.

File: acos_client/v21/action.py [hunk 3]
```diff
@@ -35,3 +35,10 @@ def reload(self, write_memory=False, **kwargs):
         write_memory = 1 if write_memory else 0
         return self._post("system.action.reload",
                           params={"write_memory": write_memory}, **kwargs)
+
+    def activate_and_write(self, partition, **kwargs):
+        if partition:
```
Comment (dougwig) [inline acos_client/v21/action.py]: Don't need this conditional if it's required

**Comment by mdurrant-b3** [comment]
Cursed 3.4....

**Pull request merged**